src/pages/Results/Results.page.jsx

```diff
@@ -1,24 +1,26 @@
 import React from 'react';
 
-import { withRouter } from 'react-router';
-import queryString from 'query-string';
+/* import { withRouter } from 'react-router';
+import queryString from 'query-string'; */
 import { Redirect } from 'react-router-dom';
 
+import { useGlobalContext } from '../../providers/Global/Global.provider';
+
 import VideoList from '../../components/VideoList/VideoList';
 
-const Results = ({ location }) => {
-  const parsedQuery = queryString.parse(location.search);
-  const searchQuery = parsedQuery.search_query;
+const Results = () => {
+  const { globalState } = useGlobalContext();
+  const { searchTerm } = globalState;
 
-  if (!searchQuery) {
+  if (!searchTerm) {
     return <Redirect to="/" />;
   }
```

Comment:
I feel like this logic could live inside `VideoList`, so you don't need to pass the `searchTerm` as a prop to it. I might be wrong, but consider the possibility.